gdb/mips-tdep.h

```diff
@@ -34,6 +34,7 @@ enum mips_abi
     MIPS_ABI_O64,
     MIPS_ABI_EABI32,
     MIPS_ABI_EABI64,
+    MIPS_ABI_CHERIABI,
```

Comment:
There are many more changes to make to do this correctly.  For expediency I did all of the CHERI-specific gdbarch setup in mips-fbsd-tdep.c (that sets gdbarch_ptr_bit, etc. for FreeBSD CheriABI binaries).  If we want to support non-FreeBSD CheriABI binaries then some of those changes (such as gdbarch_ptr_bit, but also some others such as the address_to_pointer methods, etc.) need to move from mips-fbsd-tdep.c to mips-tdep.c.  I didn't think we had any non-FreeBSD CheriABI binaries that we were running under gdb currently.   Presumably one might generate bare metal binaries and want to debug them via qemu's gdb stub in which case moving the non-OS specific bits into mips-tdep.c might make sense.